src/avram/queryable.cr

```diff
@@ -44,6 +44,13 @@ module Avram::Queryable(T)
       .select(@@schema_class.column_names)
   end
 
+  def clone : self
+    original_query = query
+    instance = self.class.new
+    instance.query.merge(original_query)
```

Comment:
I think instead of query.merge we should  have  query.clone that basically  does a merge but merges everything. Selects, orders, limits, etc. Thoughts?

Reply:
That's a good idea.